File: LongestMonotonicSubarray/longestMonotonicSubarray.py
from typing import List

def longestMonotonicSubarray(nums: List[int]) -> int:
    inc = 1
    dec = 1
    max_length = 1
    for i in range(0, len(nums) - 1):
        if nums[i] > nums[i + 1]:
            dec += 1
            inc = 1
        elif nums[i] < nums[i + 1]:
            inc += 1
            dec = 1
        else:
            inc = dec = 1
        max_length = max(max_length, inc, dec)
    return max_length

print(longestMonotonicSubarray([1,4,3,3,2]
))

File: LongestMonotonicSubarray/longestMonotonicSubarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestMonotonicSubarray = function(nums) {
    let inc = 1;
    let dec = 1;
    let max_length = 1;
    for (let i = 0; i < nums.length - 1; i++) {
        if (nums[i] > nums[i + 1]) {
            dec += 1;
            inc = 1;
        } else if (nums[i] < nums[i + 1]) {
            inc += 1;
            dec += 1;
        } else {
            inc = 1;
            dec = 1;
        }
        max_length = Math.max(max_length, inc, dec);
    }
    return max_length
};